specs/trip_dispatch_spec.rb

```diff
@@ -87,6 +87,191 @@
       driver.trips.must_include trip
       passenger.must_be_instance_of RideShare::Passenger
       passenger.trips.must_include trip
+
+    end
+
+    it 'stores start_time and end_time as time objects' do
+      dispatcher = RideShare::TripDispatcher.new
+      trip = dispatcher.trips.first
+
+      trip_starts = trip.start_time
+      trip_ends = trip.end_time
+
+      trip_starts.must_be_instance_of Time
+      trip_ends.must_be_instance_of Time
+    end
+  end
+
+  describe "Wave 2: TripDispatcher#request_trip(passenger_id)" do
+
+    it "assigns a driver to the trip" do
+      dispatcher = RideShare::TripDispatcher.new
+      result = dispatcher.request_trip(34)
+      result.must_be_instance_of RideShare::Trip
+      result.driver.wont_be_nil
+    end
+
+    # This test is superseded by later tests
+    # it "chooses the first driver whose status is :AVAILABLE" do
+    #   dispatcher = RideShare::TripDispatcher.new
+    #   drivers = dispatcher.drivers
+    #   driver_hash = Hash[drivers.collect { |driver| [driver.id, driver.status] } ]
+    #   expected_value = driver_hash.key(:AVAILABLE)
+    #
+    #   result = dispatcher.request_trip(34)
+    #
+    #   result.driver.must_be_instance_of RideShare::Driver
+    #   result.must_be_instance_of RideShare::Trip
+    # end
+
+    it "uses the current time for the start time" do
+      dispatcher = RideShare::TripDispatcher.new
+      result = dispatcher.request_trip(34)
+
+      current_time = Time.now.to_s
+      result_time = result.start_time.to_s
+
+      result_time.must_equal current_time
+
+    end
+
+    it "end date, cost and rating are nil" do
+      dispatcher = RideShare::TripDispatcher.new
+      result = dispatcher.request_trip(34)
+
+      result.end_time.must_be_nil
+      result.cost.must_be_nil
+      result.rating.must_be_nil
+    end
+
+    describe "edge cases for request_trip" do
+      before do
+        @dispatcher = RideShare::TripDispatcher.new
+      end
+      it "raises an error if there are no more available drivers" do
+        drivers = @dispatcher.drivers
+
+
+        proc{ drivers.each do |driver|
+          @dispatcher.request_trip(14)
+        end }.must_raise ArgumentError
```

Comment:
This is a clever way to implement this test - good work.

For readability, I think I would put the `proc` and `must_raise` on their own lines:

```ruby
proc {
  drivers.each do |driver|
    @dispatcher.request_trip(14)
  end
}.must_raise ArgumentError
```